iop/fs/bdmfs_fatfs/src/fs_driver.c

```diff
@@ -667,12 +679,26 @@ int fs_ioctl2(iop_file_t *fd, int cmd, void *data, unsigned int datalen, void *r
             ret = file->obj.sclust;
             break;
         case USBMASS_IOCTL_GET_LBA:
-            ret = clst2sect(file->obj.fs, file->obj.sclust);
+            // Check for a return buffer and copy the 64bit LBA. If no buffer is provided return an error.
+            if (rdata == NULL || rdatalen < sizeof(u64))
+            {
+                ret = -EINVAL;
+                break;
+            }
+
+            // Get the block device backing the file so we can get the starting LBA of the file system.
+            struct block_device* bd = fatfs_fs_driver_get_mounted_bd_from_index(file->obj.fs->pdrv);
+            
+            *(u64*)rdata = clst2sect(file->obj.fs, file->obj.sclust) + bd->sectorOffset;
+            ret = 0;
```

Comment:
Same here, why is this needed, when it was not needed before?

Reply:
See explanation for the previous comment.